src/rhsmlib/facts/hwprobe.py

```diff
@@ -722,16 +724,16 @@ def get_network_interfaces(self):
                         scope = 'global'
 
                     for mkey in ipv6_metakeys:
-                        key = '.'.join(['net.interface', info.device, 'ipv6_%s' % (mkey), scope])
-                        list_key = "%s_list" % key
                         # we could specify a default here... that could hide
                         # api breakage though and unit testing hw detect is... meh
                         attr = getattr(addr, mkey) or 'Unknown'
-                        netinfdict[key] = attr
-                        if not netinfdict.get(list_key, None):
-                            netinfdict[list_key] = str(attr)
-                        else:
-                            netinfdict[list_key] += ', %s' % str(attr)
+                        ipv6_values[mkey][scope].append(str(attr))
+                for mkey, mvalue in ipv6_values.items():
+                    for skey, svalue in mvalue.items():
+                        key = '.'.join(['net.interface', info.device, 'ipv6_%s' % (mkey), skey])
+                        list_key = "%s_list" % key
+                        netinfdict[key] = svalue[0]
```

Comment:
I thought about it, but it is not needed: each `(key, value)` mapping in `ipv6_values` (and `ipv4_values`) exists because we call `.append()` to add a new element (so defaultdict adds the new mapping if not existing). Hence, all the lists in the mappings will have at least one element.